syntax = "proto2";

// revision: r0, 2019-05-23

enum KeyType {
	RSA = 0;
	Ed25519 = 1;
	Secp256k1 = 2;
	ECDSA = 3;
}

message PublicKey {
	required KeyType Type = 1;
	required bytes Data = 2;
}

message PrivateKey {
	required KeyType Type = 1;
	required bytes Data = 2;
}

// from plaintext r0, 2019-05-27
message Exchange {
  optional bytes id = 1;
  optional PublicKey pubkey = 2;
}
